test/v2/connectors/RetrieveHistoricFhlUkPropertyAnnualSubmissionConnectorSpec.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.connectors
+
+import mocks.MockAppConfig
+import org.scalamock.handlers.CallHandler
+import v2.mocks.MockHttpClient
+import v2.models.domain.Nino
+import v2.models.errors.{ DownstreamErrorCode, DownstreamErrors }
+import v2.models.outcomes.ResponseWrapper
+import v2.models.request.retrieveHistoricFhlUkPropertyAnnualSubmission.RetrieveHistoricFhlUkPropertyAnnualSubmissionRequest
+import v2.models.response.retrieveHistoricFhlUkPropertyAnnualSubmission.{
+  AnnualAdjustments,
+  AnnualAllowances,
+  RentARoom,
+  RetrieveHistoricFhlUkPropertyAnnualSubmissionResponse
+}
+
+import scala.concurrent.Future
+
+class RetrieveHistoricFhlUkPropertyAnnualSubmissionConnectorSpec extends ConnectorSpec {
+
+  val nino: String    = "AA123456A"
+  val taxYear: String = "2019-20"
+
+  val request: RetrieveHistoricFhlUkPropertyAnnualSubmissionRequest = RetrieveHistoricFhlUkPropertyAnnualSubmissionRequest(
+    nino = Nino(nino),
+    taxYear = taxYear
+  )
+
+  val annualAdjustments = AnnualAdjustments(
+    Option(BigDecimal("100.11")),
+    Option(BigDecimal("200.11")),
+    Option(BigDecimal("105.11")),
+    true,
+    Option(BigDecimal("100.11")),
+    false,
+    Option(RentARoom(true))
+  )
+
+  val annualAllowances = AnnualAllowances(
+    Option(BigDecimal("100.11")),
+    Option(BigDecimal("300.11")),
+    Option(BigDecimal("405.11")),
+    Option(BigDecimal("550.11"))
+  )
```

Comment:
This is fine but the HttpClient mocking does not use the JSON Writes so the contents of the request body doesn't get touched and can be as simple as you like.